tools/roll.py

```diff
@@ -77,28 +85,41 @@ def rel(subdir):
     shutil.rmtree('vendor')
 
   if incl is not None:
-    if not os.path.isdir(incl):
-      print 'Warning: --incl directory %s does not exist, skipping.' % incl
-    else:
-      inclPath = os.path.abspath(incl)
-      inclParent, inclName = os.path.split(inclPath)
-
-      for (dirpath, dirnames, _) in os.walk(os.getcwd()):
-        if dirpath == inclParent:
-          # Don't descend into the included subdirectory.
-          dirnames.remove(inclName)
-        elif not inclPath.startswith(dirpath):
-          # Remove directories that aren't an ancestor of the included.
-          print 'rm subdirectory: %s' % dirpath
-          shutil.rmtree(dirpath)
+    inclPaths = []
+    inclParentToName = {}
+    for dir in incl:
+      if not os.path.isdir(dir):
+        print 'Warning: --incl directory %s does not exist, skipping.' % dir
+      else:
+        path = os.path.abspath(dir)
+        parent, name = os.path.split(path)
+        inclPaths.append(path)
+        inclParentToName[parent] = name
+
+    def inclPathStartsWith(dir):
+      for p in inclPaths:
+        if p.startswith(dir):
+          return True
+      return False
```

Comment:
How about:
```python
def inclPathStartsWith(dir):
  return any(p.startswith(dir) for p in inclPaths)
```
in fact at that point it might be more readable to just inline it when you check it.